path_oidc.go

```diff
@@ -215,13 +215,14 @@ func (b *jwtAuthBackend) authURL(ctx context.Context, req *logical.Request, d *f
 
 	config, err := b.config(ctx, req.Storage)
 	if err != nil {
-		logger.Warn("error loading configuration", "error", err)
-		return resp, nil
+		return nil, err
 	}
-
 	if config == nil {
-		logger.Warn("nil configuration")
-		return resp, nil
+		return logical.ErrorResponse("could not load configuration"), nil
```

Comment:
Just for my own education, what is the value of returning `nil, err` in some places, versus `errResp, nil` in other places? I actually don't know and I've tended toward the first.

Reply:
Broadly (as I know it), it has to do with 400s (err,nil) vs 500s (nil, err). In this case, a config being nil is something the user controls pretty directly, but the previous case of a storage error is more of a 500/"Internal Service" type error. But that's just observation of how the pattern has been used, and there is variation. But it definitely does effect whether the resp code is a 4xx or 5xx.

Reply:
Ah! That is super helpful! Thanks!